recursion/compiler/examples/fibonacci.rs

```diff
@@ -0,0 +1,34 @@
+use p3_baby_bear::BabyBear;
+use p3_field::AbstractField;
+use sp1_core::stark::StarkGenericConfig;
+use sp1_core::utils::BabyBearPoseidon2;
+use sp1_recursion_compiler::prelude::*;
+use sp1_recursion_core::runtime::Runtime;
+
+fn main() {
+    let mut builder = AsmBuilder::<BabyBear>::new();
+    let a: Felt<_> = builder.constant(BabyBear::zero());
```

Comment:
It's weird to me that Felt has a generic in F, but Bool doesn't.

Reply:
That's because we need to know how to make arithmetic operations on Felt<F>